code/datums/ai/basic_mobs/pet_commands/fetch.dm

```diff
@@ -102,16 +101,18 @@
 /datum/ai_behavior/eat_fetched_snack/perform(seconds_per_tick, datum/ai_controller/controller, target_key, delivery_key)
 	. = ..()
 	var/obj/item/snack = controller.blackboard[target_key]
-	if(QDELETED(snack) || !isturf(snack.loc) || ishuman(snack.loc))
+	if(QDELETED(snack))
+		finish_action(controller, FALSE) // Where did it go?
+	var/is_living_loc = isliving(snack.loc)
+	if(QDELETED(snack) || (!isturf(snack.loc) && !is_living_loc))
```

Comment:
we don't sleep here at all, so there's no reason to check the snack again. no way it gets qdel'd between 104 and 107.

```suggestion
	if(QDELETED(snack))
		finish_action(controller, FALSE) // Where did it go?

	var/is_living_loc = isliving(snack.loc)
	if(!isturf(snack.loc) && !is_living_loc)
```

Reply:
wait, is it because of the `finish_action()` call? Shouldn't we be returning there anyways? If something wack happens in finish_action you can ignore this review but i don't get why we're doublechecking.